### PR TITLE
🔧 MAINTAIN: Reduce test warnings

### PR DESCRIPTION
**Pull request by chrisjsewell**
This PR is focussed on reducing pytest warnings:

- Replace `collections` with `collections.abc`
- pytest-asyncio does not work with `unittest.TestCase` derived tests (https://github.com/pytest-dev/pytest-asyncio/issues/77).
- `ProcessFuture` already closed via polling should not set a result via a broadcast event.
- Upgrade kiwipy and plumpy to fix:
  - https://github.com/aiidateam/kiwipy/pull/98
  - https://github.com/aiidateam/plumpy/pull/204
  - https://github.com/aiidateam/plumpy/pull/206

partially addresses #4722

### CODE REVIEW

**Comment by chrisjsewell** [comment]
719 warnings (https://github.com/aiidateam/aiida-core/runs/1918314856), down to 123 warnings (https://github.com/aiidateam/aiida-core/pull/4742/checks?check_run_id=1918695277)

There is still a lot of:

- ResourceWarning: unclosed event loop <_UnixSelectorEventLoop running=False closed=False debug=True>
- ResourceWarning: unclosed event loop <_UnixSelectorEventLoop running=False closed=False debug=False>
- RuntimeWarning: coroutine 'create_task.<locals>.run_task' was never awaited

but these are more difficult to debug so I will tackle these in another PR.

Also @ltalirz, to note with the new SSH tests, we are getting:

- ResourceWarning: unclosed <socket.socket fd=23, family=AddressFamily.AF_INET6, type=SocketKind.SOCK_STREAM, proto=0, laddr=('::1', 55576, 0, 0)>

**Comment by chrisjsewell** [comment]
I've released plumpy v0.18.5 https://anaconda.org/conda-forge/plumpy/files, so I don't know why its not finding it

**Comment by ltalirz** [comment]
> Also @ltalirz, to note with the new SSH tests, we are getting:
> 
> * `ResourceWarning: unclosed <socket.socket fd=23, family=AddressFamily.AF_INET6, type=SocketKind.SOCK_STREAM, proto=0, laddr=('::1', 55576, 0, 0)>`

Exactly, I already opened https://github.com/aiidateam/aiida-core/issues/4744 for this

> I've released plumpy v0.18.5 https://anaconda.org/conda-forge/plumpy/files, so I don't know why its not finding it

In my experience it can take a while.

**Comment by chrisjsewell** [comment]
> In my experience it can take a while to "propagate" to every cache.

yep its installing now 👍

File: aiida/engine/processes/futures.py [hunk 1]
```diff
@@ -59,7 +59,12 @@ def __init__(
 
             # Try setting up a filtered broadcast subscriber
             if self._communicator is not None:
-                broadcast_filter = kiwipy.BroadcastFilter(lambda *args, **kwargs: self.set_result(node), sender=pk)
+
```
Comment (ltalirz) [inline aiida/engine/processes/futures.py:62]: Thanks - I was going to suggest to add a comment here as to why that is necessary but then I realized that this is in the `ProcessFuture` 

https://github.com/aiidateam/aiida-core/issues/4758

**Review by ltalirz — COMMENTED** [review]
thanks @chrisjsewell , this is great!

I still spotted one overlooked collection import
```
tests/test_dataclasses.py::TestTrajectoryData::test_conversion_from_structurelist
  /home/runner/work/aiida-core/aiida-core/aiida/orm/nodes/data/array/trajectory.py:38: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3,and in 3.9 it will stop working
    if not isinstance(symbols, collections.Iterable):
```

Happy to approve once the tests pass

File: aiida/engine/processes/futures.py (hunk 1 above)
Comment (chrisjsewell) [reply aiida/engine/processes/futures.py:62]: yeh I thought that might be the case

**Review by ltalirz — APPROVED** [review]
thanks @chrisjsewell !

**Pull request merged**